### ISSUE TITLE
Include nested template with string

### ISSUE THREAD

**Issue by iggyzuk** [rossmacarthur/upon#22]
Hi, I’m a little puzzled, when including a template can I not pass a simple string inside it using the **with** keyword? 

```rust
#[derive(Serialize)]
struct Template {
    colors: Vec<&'static str>,
}
```

Main template:
```html
{% for color in colors %} 
    {% include "color" with color %} 
{% endfor %}
```

Color template:
```html
My color is: {{ color }}
```

The error I get:
```
  --> color:1:17
   |
 1 | My color is: {{ color }}
   |                 ^^^^^
   |
   = reason: not found in this scope
```

In order to compile I must do the following but it feels verbose.
```rust
#[derive(Serialize)]
struct Template {
    colors: Vec<Color>,
}

#[derive(Serialize, Clone)]
struct Color {
    color: &'static str,
}
```

**Comment by rossmacarthur** [comment]
Currently `with {value}` makes `{value}` the _root_ context for the nested template. So the first error is because it is trying to lookup the field `color` from the string. The way you are doing it is currently the best way to do, but I agree it is a bit verbose.

You should actually not have not specify `with` at all, but it appears there is a bug with accessing loop variables in the parent template 😬. I have fixed it in d9e46eaa41c30123cafcfc08be61f966281d577d, so the following should work in the next release.

```jinja
{% for color in colors %} 
    {% include "color" %} 
{% endfor %}
```

With regards to using `with` I can see a few potential improvements here:

**Improve passing of value into the nested template**

- We could support map literals and allow constructing the literal inline before passing, for example
    ```jinja
    {% include "color" with { color: color } %}
    ```
    This is still a bit verbose

**Improve referring to the root context**

- We could support referring to the current root context with `.`
    ```jinja
    My color is: {{ . }}
    ```

**Comment by rossmacarthur** [comment]
Fix is released, I'm going to close this issue, but I have opened #23, and #24
